docs/compass/03-packages-credential-requests.md

```diff
@@ -0,0 +1,136 @@
+1. Application sets credentials for a given Service Instance.
+1. Runtime Agent fetches configuration with credentials for a given Service Instance.
+
+### GraphQL Schema
+
+```graphql
+type Package {
+  id: ID!
+  # (...)
+
+  """
+  Optional JSON schema for validation user input while provisioning Service Class.
+  """
+  instanceAuthRequestInputSchema: JSONSchema
+  instanceAuth(id: ID!): PackageInstanceAuth
+  instanceAuths: [PackageInstanceAuth!]!
+  """
+  When defined, all requests via `requestPackageInstanceAuthCreation` mutation fallback to defaultInstanceAuth.
+  """
+  defaultInstanceAuth: Auth
+}
+
+type PackageInstanceAuth {
+  id: ID!
+  """
+  Context of PackageInstanceAuth - such as Runtime ID, namespace, etc.
+  """
+  context: Any
+
+  """
+  User input
+  """
+  inputParams: Any
+  
+  """
+  It may be empty if status is PENDING.
+  Populated with `package.defaultInstanceAuth` value if `package.defaultAuth` is defined. If not, Compass notifies Application/Integration System about the Auth request.
+  """
+  auth: Auth
+  status: PackageInstanceAuthStatus
+}
+
+type PackageInstanceAuthStatus {
+  condition: PackageInstanceAuthStatusCondition!
+  timestamp: Timestamp!
+  message: String!
+  """
+  Possible reasons:
+  - PendingNotification
+  - NotificationSent
+  - CredentialsProvided
+  - CredentialsNotProvided
+  - PendingDeletion
+  """
+  reason: String!
+}
+
+enum PackageInstanceAuthStatusCondition {
+  """
+  When creating, before Application sets the credentials
+  """
+  PENDING
+  SUCCEEDED
+  FAILED
+  """
+  When Runtime requests deletion and Application has to revoke the credentials
+  """
+  UNUSED
+}
+
+input PackageInstanceAuthRequestInput {
+	"""
+	Context of PackageInstanceAuth - such as Runtime ID, namespace, etc.
+	"""
+	context: Any
+	"""
+	JSON validated against package.instanceAuthRequestInputSchema
+	"""
+	inputParams: Any
+}
+
+type Mutation {
+  """
+  When PackageInstanceAuth is not in pending state, the operation returns error.
+
+  When used without error, the status of pending auth is set to success.
+  """
+  setPackageInstanceAuth(packageID: ID!, authID: ID!, in: AuthInput! @validate): PackageInstanceAuth!
+  deletePackageInstanceAuth(packageID: ID!, authID: ID!): PackageInstanceAuth!
+  requestPackageInstanceAuthCreation(packageID: ID!, in: PackageInstanceAuthRequestInput!): PackageInstanceAuth!
+  requestPackageInstanceAuthDeletion(packageID: ID!, authID: ID!): PackageInstanceAuth!
+}
+```
+
+Application or Integration System can set optional `instanceAuthRequestInputSchema` field with a JSON schema with parameters needed during Service Class provisioning. The values provided by User are validated against the JSON schema.
+
+### Example request credentials flow
+
+1. User connects Application `foo` with single Package `bar` which contain few API and Event Definitions. The Package has `instanceAuthRequestInputSchema` defined.
```

Comment:
Here we have duplicated information, see line 26.
Also, points are not displayed properly.